drv/stm32h7-hash-server/src/main.rs

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! STM32H7 HASH server.
+//!
+//! This server is responsible for managing access to the HASH engine.
+//!
+
+#![no_std]
+#![no_main]
+
+// use core::convert::TryInto;
+use userlib::*;
+
+use drv_stm32h7_hash::Hash;
+use drv_stm32xx_sys_api as sys_api;
+use idol_runtime::{ClientError, Leased, LenLimit, RequestError, R};
+
+#[cfg(feature = "h753")]
+use stm32h7::stm32h753 as device;
+
+use drv_hash_api::HashError;
+
+task_slot!(SYS, sys);
+
+const IRQ_MASK: u32 = 1;
+
+fn hash_hw_reset() {
+    let sys = sys_api::Sys::from(SYS.get_task_id());
+    sys.enter_reset(sys_api::Peripheral::Hash);
+    sys.disable_clock(sys_api::Peripheral::Hash);
+    hl::sleep_for(1);
```

Comment:
Maybe this is common in embedded code, but using sleeps feels icky to me. Will `sleep_for` always deterministically satisfy the purpose it is being used for here?